SYCL/ESIMD/force_stateless_mem.cpp

```diff
@@ -0,0 +1,422 @@
+//==------- force_stateless_mem.cpp  - DPC++ ESIMD on-device test ---------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl -fsycl-esimd-force-stateless-mem %s -o %t.out
```

Comment:
For better coverage, my suggestion would be adding compilation with `-fsycl-esimd-force-stateless-mem` to all existing accessor-based tests. I.e. all of them would contain:
```
// RUN: %clangxx -fsycl %s -o %t.out
// RUN: %clangxx -fsycl -fsycl-esimd-force-stateless-mem %s -o %t_stateless.out
...
// RUN: %GPU_RUN_PLACEHOLDER %t.out
// RUN: %GPU_RUN_PLACEHOLDER %t_stateless.out
```
The same could be done for this test.
Does this make sense?

Reply:
That would be less work for me and more testing coverage.
I did not do that because:
- The current tests already take longer time then recommended. (and compilation takes the most of test time in CI)
- Fail of such combined test due to stateless would cause stop of testing of non-stateless functionality as well.

Reply:
@kbobrovs - thank you for the comment. I re-worked the PR. Now it splits existing tests using accessor-based API to pairs of tests: the standard-accessor-based and accessor-based-converted-to-stateless.
Please review the tests.